_ml-commons-plugin/gpu-acceleration.md

```diff
@@ -0,0 +1,304 @@
+PYTORCH_NEURON_LIB_PATH=~/pytorch_venv/lib/python3.7/site-packages/torch_neuron/lib/
+mkdir -p $OPENSEARCH_HOME/lib/torch_neuron; cp -r $PYTORCH_NEURON_LIB_PATH/ $OPENSEARCH_HOME/lib/torch_neuron
+export PYTORCH_EXTRA_LIBRARY_PATH=$OPENSEARCH_HOME/lib/torch_neuron/lib/libtorchneuron.so
+echo "export PYTORCH_EXTRA_LIBRARY_PATH=$OPENSEARCH_HOME/lib/torch_neuron/lib/libtorchneuron.so" | tee -a ~/.bash_profile
+# Increase JVm stack size to >=2MB
+echo "-Xss2m" | tee -a $OPENSEARCH_HOME/config/jvm.options
+# Increase max file descriptors to 65535
+echo "$(whoami) - nofile 65535" | sudo tee -a /etc/security/limits.conf
+# max virtual memory areas vm.max_map_count to 262144
+sudo sysctl -w vm.max_map_count=262144
+```
+
+OpenSearch should now be running inside your GPU-accelerated cluster. However, if any errors occur during provisioning, you can install the GPU accelerator drivers manually.
+
+## (AWS Inferentia) Installing GPU accelerator drivers manually
+
+If the previous two scripts do not provision your GPU-accelerated node properly, you can install the drivers for AWS Inferentia manually:
+
+1. Deploy an AWS accelerator instance based on your chosen Linux operating system. For instructions, see [Deploy on AWS accelerator instance](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuron/setup/pytorch-install.html#deploy-on-aws-ml-accelerator-instance).
+
+2. (Optional) To monitor the GPU usage of your accelerator instance, install [Neuron tools](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/tools/index.html), which allows models to be used inside your instance:
+
+   ```
+   # Install Neuron Tools
+   sudo apt-get install aws-neuronx-tools -y
+   ```
+
+   ```
+   # Add Neuron tools your PATH
+   export PATH=/opt/aws/neuron/bin:$PATH
+   ```
+  
+   ```
+   # Test Neuron tools
+   neuron-top
+   ```
+
+4. Copy the Neuron library into OpenSearch. The following command uses a directory named `opensearch-2.5.0`:
+
+   ```
+   OPENSEARCH_HOME=~/opensearch-2.5.0
+   ```
+
+5. Activate  the `pytorch` virtual environment:
+
+   ```
+   source pytorch_venv/bin/activate
+   ```
+ 
+6. Set the `PYTORCH_NEURON_LIB_PAT` path. In this example, we create a `pytorch` virtual environment in the OPENSEARCH_HOME folder:
+
+   ```
+   PYTORCH_NEURON_LIB_PATH=~/pytorch_venv/lib/python3.7/site-packages/torch_neuron/lib/
+
+
+   mkdir -p $OPENSEARCH_HOME/lib/torch_neuron; cp -r  $PYTORCH_NEURON_LIB_PATH/ $OPENSEARCH_HOME/lib/torch_neuron
+   export PYTORCH_EXTRA_LIBRARY_PATH=$OPENSEARCH_HOME/lib/torch_neuron/lib/libtorchneuron.so
+  ```
+
+7. To make sure you have enough memory to upload a model, increase the JVM stack size to `>+2MB`:
+
+   ```
+   echo "-Xss2m" | sudo tee -a $OPENSEARCH_HOME/config/jvm.options
+   ```
+
+8. Start OpenSearch. 
+
+### Troubleshooting
+
+Due to the high memory required to load models, you might encounter the following error when trying to run OpenSearch in a GPU-accelerated cluster: 
```

Comment:
These error is not from high memory usage, it's for file descriptor.

Reply:
Why are we setting the `vm.max_map_count` in the second command then?

Reply:
I mean the first error is for file descriptor.